src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,139 @@ def standardize_patient_name(
+                         is from.
+
+        country_extractor (Callable): A function to parse a provided resource that
+                                      yields a generator to return the standard 2 letter
+                                      abreviation of each country found in the resource
+                                      when itterated over.
+    Returns:
+        standardized (str): The standardized phone number in E.164 format when the raw
+                            phone number was succesfully parsed and an emptry string
+                            otherwise.
     """
 
-    raw = [x for x in raw if x.isnumeric()]
-    raw = "".join(raw)
-    if len(raw) != 10:
-        raw = None
-    return raw
+    phone_number = None
+
+    # First Attempt - Assume complete international format provided.
+    try:
+        phone_number = phonenumbers.parse(raw, None)
+    except phonenumbers.phonenumberutil.NumberParseException:
+
+        # Second Attempt - If a resource and country extraction function have been
+        #                  provided attempt to parse using countries found in resource.
+        if resource is not None and country_extractor is not None:
+            for country in country_extractor(resource):
+                try:
+                    phone_number = phonenumbers.parse(raw, country)
+                    break
+                except phonenumbers.phonenumberutil.NumberParseException:
+                    continue
+
+        # Third Attempt - Assume the phone number is American.
+        if phone_number is None:
+            try:
+                phone_number = phonenumbers.parse(raw, "US")
+            except phonenumbers.phonenumberutil.NumberParseException:
+                standardized = ""
+
+    if phone_number is not None:
+        standardized = str(
+            phonenumbers.format_number(
+                phone_number, phonenumbers.PhoneNumberFormat.E164
+            )
+        )
+
+    return standardized
+
+
+def standardize_country(
+    country: str, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
+) -> str:
+    """
+    Given a country return it in a standard form as specified by code_type.
+
+    Args:
+        country (str): Country to be standardized.
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+
+    Returns:
+        (str): Country in standardized form, or None if unable to standardize.
+    """
+    if len(country) == 3 and country.isalpha():
+        country = pycountry.countries.get(alpha_3=country)
+    elif len(country) == 2 and country.isalpha():
+        country = pycountry.countries.get(alpha_2=country)
+    elif len(country) == 3 and country.isnumeric():
+        country = pycountry.countries.get(numeric=country)
+    elif len(country) > 4 and country.isalpha:
+        possible_countries = pycountry.countries.search_fuzzy(country)
```

Comment:
I don't think we should be adding in fuzzy matching for countries. Especially not with a way to set a threshold, and certainly not without indicating in the doc string that it's an option.

Reply:
Looking at the code, I don't think there is a threshold option for fuzzy matching.  Maybe instead you could accept a list or tuple of input formats to check for?
```
input_formats: Iterable[str]
```
input_formats could be defaulted to ("alpha_2", "alpha_3", "numeric", "fuzzy") in the function, and your cases would check for presence of the appropriate value before trying that kind of conversion.

Also, change `code_type` to `output_format`

Reply:
Very much agree to no fuzzy matching for this. I think as a general rule we should save anything that might use fuzzy matching for post 6/30.